### PR TITLE
Allow getting BakedModels using Identifiers

### PR DESCRIPTION
**Pull request by PepperCode1**
If a model is loaded using an `ExtraModelProvider`, there is no way to retrieve it from the `BakedModelManager` after it is baked because it does not have a corresponding `ModelIdentifier`. This pull request adds a `ModelHelper` that allows retrieving a model via `Identifier` instead of `ModelIdentifier`. This works by exposing the full functionality of `BakedModelManager`'s internal model map, which stores `Identifier`s as keys, even though vanilla's retrieval method requires a `ModelIdentifier`. The same `Identifier` used to load the model can be used in `ModelHelper#getModel` to get the baked model.

### CODE REVIEW

File: fabric-models-v0/src/main/java/net/fabricmc/fabric/api/client/model/ModelHelper.java
Comment (liach) [inline fabric-models-v0/src/main/java/net/fabricmc/fabric/api/client/model/ModelHelper.java]:
```suggestion
	 * @param manager the manager that holds models
	 * @param id the id of the model
	 * @return the model
```

**Review by liach — APPROVED** [review]

**Review by Technici4n — COMMENTED** [review]
Looking good, will test in a bit!

What are the guarantees of this wrt. `bake` ordering? Is it guaranteed that model dependencies are already baked and can be safely retrieved using `getModel`, and would it be worth mentioning this in the javadoc?

Other small thing: `fabric-renderer-v1-api` already has a `ModelHelper`, it's probably not a big deal though, and `import static` can be used in case of conflicts. Thoughts?

**Comment by PepperCode1** [comment]
The BakedModel map is retrieved all at once from the ModelLoader in `BakedModelManager#apply`, so this method (as well as the vanilla method) should only be used after the `BakedModelManager` has completed reloading.

I feel like there could be cases where both `ModelHelper`s might need to be used in the same class. Would a better name be `BakedModelHelper` or `ModelRetrievalHelper`, or maybe something else?

**Comment by liach** [comment]
Imo finding a different name for helper class isn't the good way. #1366 should be the way to go, i.e. merging modules that are unnecessarily split.

**Comment by PepperCode1** [comment]
I'm fine with merging this class, but does that mean this PR will have to wait until the module PR is merged?

**Comment by OroArmor** [comment]
go ahead and work on this pr. if #1366 gets merged (which it will probably not) this can be changed, or if this is merged before, #1366 will make the changes to reduce work for you

**Review by LambdAurora — APPROVED** [review]

**Comment by Technici4n** [comment]
> Imo finding a different name for helper class isn't the good way. #1366 should be the way to go, i.e. merging modules that are unnecessarily split.

No. This class is about baked model loading, the other one is about baked model rendering, they should be separate. `ModelLoadingHelper` maybe?

**Comment by LambdAurora** [comment]
> > Imo finding a different name for helper class isn't the good way. #1366 should be the way to go, i.e. merging modules that are unnecessarily split.
> 
> No. This class is about baked model loading, the other one is about baked model rendering, they should be separate. `ModelLoadingHelper` maybe?

Doesn't look very loading to me, the only method present in it is called after all loading is done.

**Comment by Technici4n** [comment]
> The BakedModel map is retrieved all at once from the ModelLoader in `BakedModelManager#apply`, so this method (as well as the vanilla method) should only be used after the `BakedModelManager` has completed reloading.

This would be worth adding to the javadoc imo. Trying to get another baked model during `bake()` is very tempting. :-P

> I feel like there could be cases where both `ModelHelper`s might need to be used in the same class.

Yeah.

**Comment by Technici4n** [comment]
`BakedModelManagerExtensions`?

**Comment by LambdAurora** [comment]
> `BakedModelManagerExtensions`?

`BakedModelManagerHelper` maybe? Would be consistent with resource loader with its `ResourceManagerHelper` but considering it's a v0 module I don't really know.

**Comment by liach** [comment]
Maybe like `Models` or `BakedModels`

**Comment by Technici4n** [comment]
`BakedModelManagerHelper` is good I think.

**Comment by Technici4n** [comment]
image
Coolest test mod ever!!!! 🤠

**Review by Technici4n — APPROVED** [review]

**Review by Player3324 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
